java/com/puppetlabs/trapperkeeper/services/webserver/jetty9/utils/InternalSslContextFactory.java

```diff
@@ -0,0 +1,84 @@
+package com.puppetlabs.trapperkeeper.services.webserver.jetty9.utils;
+
+import org.eclipse.jetty.util.log.Log;
+import org.eclipse.jetty.util.log.Logger;
+import org.eclipse.jetty.util.security.CertificateUtils;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+
+import java.io.File;
+import java.security.cert.CRL;
+import java.util.Collection;
+import java.util.function.Consumer;
+
+public class InternalSslContextFactory extends SslContextFactory {
+
+    private static int maxTries = 25;
+    private static int sleepInMillisecondsBetweenTries = 100;
+    private static final Logger LOG =
+            Log.getLogger(InternalSslContextFactory.class);
+    private static Consumer<SslContextFactory> consumer = sslContextFactory
+            -> {};
+
+    private Collection<? extends CRL> _crls;
+
+    @Override
+    protected Collection<? extends CRL> loadCRL(String crlPath) throws Exception {
+        Collection<? extends CRL> crls;
+
+        synchronized (this) {
+            if (_crls == null) {
+                crls = super.loadCRL(crlPath);
+            } else {
+                crls = _crls;
+            }
+        }
+
+        return crls;
+    }
+
+    public void reload() throws Exception {
```

Comment:
I thought this was what was already built into Jetty?

Reply:
No, the version in Jetty takes and argument, you can see us calling it at the end of this function.

Reply:
Oh never mind, I see the difference now. That branch had substantially more revision to backport the synchronized state mgmt around the SSL content.

Reply:
Yeah that's the conclusion I arrived at too.

Reply:
yeah and it looks like this version is an improved adaptation to allow us to do more load attempts if something is incorrect with the crl itself